### DISCUSSION TITLE
Does wiki.js support SSO redirection please?

### DISCUSSION THREAD

**Discussion by biganan2012** [requarks/wiki#4317] [Help / Questions]
BackGround:
I am looking for a wiki tool for my company and after some investigation, wiki.js is an option because of the awesome UI. Also i see most contributions here.

We also have 2 other systems. Let's say A and B here.
A: A system which can do SSO to other apps. Here SSO only stand for authentication.
A add a link for wiki.js. When user login to A, user gets a token and userId. A request like http:///access_token=xxxx&userId=yyyy generated to App.
App should callback with url like http://<A_IP>/
zuul/data-auth-provider/open_api/user/get_user_by_id?accesstoken=xxxx&userId=yyyy. When authenticated success, A system will return confirmation code and user information to App.
Here system A has no aothorization.

B: B system which can do SSO and called with SSO center. SSO here means authentication and authorization.
B add a link for wiki.js. When user login to B, user gets a token. Then app generate request like http:///access_token=xxxx&appId=yyyy to B. Here appId used for authorization. A user might be administrator in app X but normal user in app Y, so need different appId to distinguish accordingly.

You may say SSO means we can login through any app to SSO center. But now we just talk about this scenario, to redirect to wiki.js from SSO center after login.

Need Help:
Does wiki.js support with such integration with system A or B please? I see we have authentications like OAuth, but not for SSO.
Could some one figure it out please? Thank you very much!

**Comment by NGPixel** [comment]
No this is not supported at the moment.

**Reply by biganan2012** [reply]
Thank you for your reply. If I want to implement this, how could I get started please?
The basic requirement is just to login to system A, and then click a button or icon that using token in URL (like http://<nodejs_ip>/access_token=xxx&userId=yyyy ) to hide the login form.
I have went through the developer guide, seems i need to write a new module here? I am really new to this, so any comments will be very appreciated. Thank u very much.
